libraries/botbuilder-core/src/turnContext.ts

```diff
@@ -119,6 +120,17 @@ export type DeleteActivityHandler = (
 
 export const BotCallbackHandlerKey = 'botCallbackHandler';
 
+function getAppropriateReplyToId(source: Partial<Activity>): string | undefined {
+    if (
+        source.type !== ActivityTypes.ConversationUpdate ||
+        (source.channelId !== Channels.Directline && source.channelId !== Channels.Webchat)
+    ) {
+        return source.id;
+    }
+
+    return undefined;
+}
+
```

Comment:
Note: I debated about what to do with this function, which is also located in `ActivityEx`. Since `TurnContext` and `ActivityEx` are in two separate libraries, I copy/pasted the function to keep exposed surface area lower. I wanted a function just to avoid the harder-to-read ternary (used in the .NET PR). So, this isn't DRY, but is DRY-er than copy/pasting the ternary around, I think.

I think this *could* be removed from here, entirely, though. In .NET, TurnContext doesn't have a `getConversationReference` and instead relies on the one in `ActivityEx`. I *think* it might be worth removing here as "Find all References" only returns it being used in 6 different places. However, this is core enough to the SDK and a little beyond the scope of this PR, so I was more hesitant to mess with it.